### ISSUE TITLE
Running Mosquitto in Windows container fails

### ISSUE THREAD

**Issue by rajeshanayaka** [eclipse/mosquitto#1071]
Running Mosquitto 1.5 64 bit with Win64 Open SSL in Windows container , mosquitto.exe is not starting at all no error being displayed as well

**Comment by Laro88** [comment]
Which mosquitto, 1.5.4? Is is running old normal windows or in a windows container in docker?
If is it normal windows then please post the output of mosquitto.exe -v (-v is for verbose)
You migmt me missing the openssl installer, it is described in the readme.

**Comment by rajeshanayaka** [comment]
Yes I am using Mosquitto 1.5.4 X64 , Windows 10 - 1803 with Docker for Windows version 18.09.0 stable 
Yes I am using with -v (verbose) switch. 

In Local computer its works fine. 
However when is run it on Windows container (microsoft/windowsservercore) its not working. 

Previous versions are also not working in container. 
I have ensured Win64 Open SSL 1.1.0 binaries and pthread (x64) binaries are copied to mosquitto directory.

**Comment by rajeshanayaka** [comment]
linux support not enabled in my windows :

PS C:\> docker run eclipse-mosquitto
Unable to find image 'eclipse-mosquitto:latest' locally
latest: Pulling from library/eclipse-mosquitto
C:\Program Files\Docker\docker.exe: no matching manifest for unknown in the manifest list entries.

**Comment by Laro88** [comment]
I have never seen that one before, normally linux images works fine on windows boxes.
Can you post your procedure for putting mosquitto on the windowsservercore, then i can try it out on my docker?

**Comment by desertspider** [comment]
If got the same problem;

Buildfile;
```
FROM mcr.microsoft.com/windows/servercore:1809

#USER ContainerAdministrator

COPY mosquitto\* /mosquitto

ENTRYPOINT CD c:/mosquitto && mosquitto -v
```

The folder mosquitto is the folder i copied from program files.

**Comment by desertspider** [comment]
Fixen by adding;
ADD https://aka.ms/vs/15/release/VC_redist.x64.exe /apache/vc_redist.x64.exe
RUN C:\apache\vc_redist.x64.exe /quiet /install

**Comment by desertspider** [comment]
And for nanoserver;

```
FROM mcr.microsoft.com/windows/nanoserver:1809

COPY vcruntime140.dll /Windows/System32/vcruntime140.dll

COPY mosquitto\* /mosquitto

ENTRYPOINT CD c:/mosquitto && mosquitto -v
```

**Comment by ralight** [comment]
Can this be closed now then?

**Comment by corelgott** [comment]
Has anybody had luck running mosquitto in nanoserver with protocol websockets?
When I run the server with protocol mqtt, everything works fine, but as soon as I switch to `protocol websockets` it just repeats the error `1559649062: Error in poll: No such file or directory.` for ever...
